row_025_views/row_025_views_safe.py

```
@@ -0,0 +1,778 @@
+                return JsonResponse({"error": f"User '{username}' not found."}, status=404)
+        else:
+            items_query = Item.objects.all()
+            users_filtered = False
+
+        # Only valid, unused, and non-expired items are used for transaction-based value calc
+        items_with_transaction_values = (
+            items_query.filter(is_used=False, expiry_date__gte=now())
+            .annotate(
+                transaction_total=Sum('transactions__value', default=0)
+            )
+            .annotate(net_value=ExpressionWrapper(F('value') + F('transaction_total'), output_field=models.DecimalField()))
```

Ruff (0.14.2)

`models` may be undefined, or defined from star imports

(F405)